client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -105,6 +105,35 @@ export const useCommandsArrayWpcom = ( {
 	const commandNavigation = useCommandNavigation( { navigate, currentRoute } );
 	const dispatch = useDispatch();
 
+	const navigateWithinSamePage = (
+		targetPath: string,
+		elementId: string,
+		commandParams: Pick< CommandCallBackParams, 'close' | 'command' >
+	) => {
+		const currentPath = window.location.pathname;
+		const targetUrl = new URL( targetPath, window.location.origin );
+
+		// Check if the user is on the same page but is not looking at the right section
+		// If not, scroll to the right section
+		if ( currentPath === targetUrl.pathname && window.location.hash !== `#${ elementId }` ) {
+			//Offset by the height of the navigation header from the top of the page
+			const fixedHeaderHeight = 72;
+			const element = document.getElementById( elementId );
+			if ( element ) {
+				const elementPosition = element.getBoundingClientRect().top + window.scrollY;
+				const offsetPosition = elementPosition - fixedHeaderHeight;
+
+				window.location.hash = elementId;
+				commandParams.close();
+				window.scrollTo( {
+					top: offsetPosition,
+					behavior: 'smooth',
+				} );
+			}
+		} else {
+			commandNavigation( targetUrl.href )( commandParams );
```

Comment:
Can we use the original target path `targetPath` ?
Using the `targetUrl.href` it reloads the whole page for me, instead of using the "react" navigation.

```suggestion
			commandNavigation( targetPath )( commandParams );
```

Reply:
Sounds good! Updated in https://github.com/Automattic/wp-calypso/pull/88565/commits/c6c96fc611d613c377c4ebf0a47ab1efdbfc7e1d